gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/SubscriptionTrustStoreLoaderManager.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.security.core;
+
+import io.gravitee.gateway.api.service.Subscription;
+import io.gravitee.node.api.certificate.KeyStoreEvent;
+import io.gravitee.node.api.server.ServerManager;
+import io.gravitee.node.vertx.server.VertxServer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+public class SubscriptionTrustStoreLoaderManager {
+
+    private final Map<String, SubscriptionTrustStoreLoader> subscriptionTrustStoreLoaders = new ConcurrentHashMap<>();
+    private final ServerManager serverManager;
+
+    public SubscriptionTrustStoreLoaderManager(ServerManager serverManager) {
+        this.serverManager = serverManager;
+    }
+
+    public void registerSubscription(Subscription subscription) {
+        final SubscriptionTrustStoreLoader loader = new SubscriptionTrustStoreLoader(subscription);
+        if (subscriptionTrustStoreLoaders.containsKey(subscription.getId())) {
+            log.debug("A TrustStoreLoader for subscription {} is already registered", subscription.getId());
+            return;
+        }
+        log.debug("Registering TrustStoreLoader for subscription {}", subscription.getId());
+        serverManager
```

Comment:
I understand that it is simpler and we can't set this in the UI. But I don't the part where we need to register it in all servers. The API model contains a server id... in my PoC I made the effort, so the code should be the same, probably some test there too. But I get it... simpler.

Reply:
see [37f0082](https://github.com/gravitee-io/gravitee-api-management/pull/8843/commits/37f008251cf80eaec2b3f229c33f9c0e2d25b9d7)